lib/client.js

```diff
@@ -21,6 +21,10 @@ var defaultConnectOptions = {
   protocolId: 'MQTT',
   protocolVersion: 4,
   reconnectPeriod: 1000,
+  backOffStrategy: null,
```

Comment:
I think `backOffStrategy` should be a `function`, rather than a string, and it should be more modular.
I'm happy if we provide this back off strategy as a separate module as well.